.github/workflows/ios.yml

```diff
@@ -19,10 +19,10 @@ jobs:
         run: |
           xcode-select --print-path
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Build
         run: |
-          set -o pipefail && xcodebuild build -scheme maplibre-navigation-ios -destination 'platform=iOS Simulator,OS=17.0.1,name=iPhone 15 Pro' | xcbeautify --renderer github-actions
+          set -o pipefail && xcodebuild build -scheme maplibre-navigation-ios -destination 'platform=iOS Simulator,OS=17.0.1,name=iPhone 15 Pro' -skipPackagePluginValidation | xcbeautify --renderer github-actions
       - name: Test
         run: |
-          set -o pipefail && xcodebuild test -scheme maplibre-navigation-ios -destination 'platform=iOS Simulator,OS=17.0.1,name=iPhone 15 Pro' | xcbeautify --renderer github-actions
\ No newline at end of file
+          set -o pipefail && xcodebuild test -scheme maplibre-navigation-ios -destination 'platform=iOS Simulator,OS=17.0.1,name=iPhone 15 Pro' -skipPackagePluginValidation | xcbeautify --renderer github-actions
```

Comment:
It's a bit unfortunate that we need this now, but OK, I guess this is required for swiftlint

Reply:
I've used SwiftLintPlugin which only contains the binary. We can of course always use SwiftLint repo and build it from source, I think this would eliminate this option, however it would significantly increase build time on our CI and also add various unrelated packages (like SourceKitten, ...) to users of this package

Reply:
Sorry for being a bit hesitant here. Could we maybe step one step back and discuss if we need the SwiftLintPlugin. I am still not 100% sure we need this plugin.

For CI we could install SwiftLint through brew or we could install it through SPM. Would this change that much about the CI build time?

While looking at other repos that use SwiftLint I stumbled over the repo from @ianthetechie here: https://github.com/stadiamaps/maplibre-swiftui-dsl-playground

They seem to use the brew install method. I am open to discussing different options, but if we can use widely used standard tools in favour of a fork of a plugin, that requires also to change our build settings, then I am leaning towards going down the standard route.

Reply:
Actually, funny that you notice that @boldtrn. A few comments...

1. We actually use swiftformat rather than swiftlint. These are quite similar projects. I don't actually have much of an opinion on preferring one or the other (or both) for a project. We just happen to use swiftformat.
2. Aside: swiftformat is actually installed in the base GitHub runners. The main reason we run `brew install` is because we have no idea when GitHub will update it 😂 So to simplify things, we just have an internal policy of running the latest and using `brew install` in Actions.
3. I agree I'd prefer to minimize use of plugins and (this is *purely* personal preference) tend to go for CI checks over commit hooks as they are significantly more robust. Mostly this is because it's *usually* simpler, more robust, and has fewer potential security risks. Just my opinion.

Reply:
I have worked in multiple companies now and it was de-facto standard to use SwiftLint. The reason I've added SwiftLint to Xcode and not actions was, that you see the warnings right away, instead of waiting for the CI to finish.

SwiftFormat can work in tandem with SwiftLint which is even encouraged, as it can automatically fix issues, the more complex ones needs to be fixed by the programmer.

If we move this to CI, how are we communicating back the warnings it found? Thats an unsolved issue right now, you would need to manually go to the checks tab and then click on summary to see the warnings - they don't show up in the PR. 

![Bildschirmfoto 2024-04-17 um 18 09 12](https://github.com/maplibre/maplibre-navigation-ios/assets/18739004/6f71dbe5-cb33-4bef-9c0c-37d5a60055af)

Of course we can fail the build if a warning is detected, but for this we first need to resolve the existing ones, where we also don't have a clear path. I've already silenced quite a lot of warnings SwiftLint found as they need quite big changes which we want to do in separate PRs anyway.